### ISSUE TITLE
Low-Light Experimental settings

### ISSUE THREAD

**Issue by Allen5123** [nachifur/RDDM#10]
您好，請問Low-Light Experimental settings 中的 Loss = L_{res}與Sampleing method = SM-Res該如何設置?
目前嘗試以下配置會error
objective = "pred_res"
test_res_or_noise = "res"

<img width="539" alt="image" src="https://github.com/nachifur/RDDM/assets/48623870/3d0be119-ee5b-4e11-ae5d-cc2858a9c2ef">

**Comment by nachifur** [comment]
您可以先自行修改和调试一下代码。我们正在逐步地整理和更新代码。

**Comment by Allen5123** [comment]
謝謝，這部分有注意到，會持續關注後續更新

**Comment by Allen5123** [comment]
您好，想請問一些deraining任務的配置 : 
1. sampling均為res + noise 如下 ?
    objective = 'pred_res_noise'
    test_res_or_noise = "res_noise"
2. convert_to_ddim 需要改成False，才會套用Eq(14)的scheduling ?
3. 原文中P(1-x,1)對應的是linear decreasing還是P(x,1) ?
![image](https://github.com/nachifur/RDDM/assets/48623870/c5b56828-2521-4df2-b2a4-ae56b406a99f)
4.目前source code中的alphas / beta的schedule是分別對應到Image restoration中的P(1-x,1)與P(x,1)嗎，還是需要另作更改 ?
![image](https://github.com/nachifur/RDDM/assets/48623870/382a5733-a910-42ec-9723-27f030ea3b60)
5. Training set是只使用Raindrop/train中的861張image ?

**Comment by nachifur** [comment]
@Allen5123 

1. Yes, for image restoration, convert_to_ddim=False.

2. For sampling, `L_{res} + L_{\epsilon}` (one Adam) needs to modify the loss function. The current code is `L_{res}, L_{\epsilon}` (two RAdma).

3. For P(1-x,1), see: https://github.com/nachifur/RDDM/issues/12#issuecomment-2002988560

4. only Raindrop/train for training.

**Comment by Allen5123** [comment]
您好，想請問關於image restoration任務的問題:
1. 如果已將line 640改為convert_to_ddim=False，是否也須一併更改line 726至convert_to_ddim=False ? 
2. 承上，Testing時，line 771 ~ 781這邊的設置與訓練時，使用的alpha, beta scheduling(__init__內初始化的值)不一樣，理論上，它們應該套用同樣的alpha, beta scheduling ? (e.g.相同的sum_scale, ratio, schedule)

**Comment by nachifur** [comment]
> 您好，想請問關於image restoration任務的問題:
> 
> 1. 如果已將line 640改為convert_to_ddim=False，是否也須一併更改line 726至convert_to_ddim=False ?
> 2. 承上，Testing時，line 771 ~ 781這邊的設置與訓練時，使用的alpha, beta scheduling(__init__內初始化的值)不一樣，理論上，它們應該套用同樣的alpha, beta scheduling ? (e.g.相同的sum_scale, ratio, schedule)

是的，convert_to_ddim = False [in L726](https://github.com/nachifur/RDDM/blob/11d06d5f389e3953fdedcf33fffbb81ff4d1583a/src/residual_denoising_diffusion_pytorch.py#L726C9-L726C31)

**Comment by Allen5123** [comment]
Thanks~
